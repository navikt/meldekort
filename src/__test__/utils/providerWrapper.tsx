import * as React from 'react';
import { store } from '../../app/store/configureStore';
import { IntlProvider } from 'react-intl-redux';
import { Provider } from 'react-redux';

interface Props {
  children: React.ReactNode;
}

const ProviderWrapper: React.FunctionComponent<Props> = props => {
  return (
    <Provider store={store}>
      <IntlProvider locale="nb" defaultLocale="nb">
        {props.children}
      </IntlProvider>
    </Provider>
  );
};

export default ProviderWrapper;
