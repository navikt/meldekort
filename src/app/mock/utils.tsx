export function erLocalhost() {
  const host: string = window.location.host;
  return host.includes('localhost') || host.includes('127.0.0.1');
}

export function erMock(): boolean {
  let meldekort = 'meldekort';
  window[meldekort] = {
    MELDEKORTSESSIONSTORAGE_USERNAME: 'username',
    MELDEKORTSESSIONSTORAGE_PASSWORD: 'password',
  };

  return process.env.REACT_APP_MOCK_FULL === 'true';
}
