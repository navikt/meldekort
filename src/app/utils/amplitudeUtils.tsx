import amplitude from 'amplitude-js';
import Environment from './env';

let initialized = false;

function initAmplitude() {
  const { amplitudeKey, amplitudeUrl } = Environment();

  if (!amplitudeKey || !amplitudeUrl) {
    return;
  }

  const config = {
    apiEndpoint: amplitudeUrl,
    saveEvents: true,
    includeUtm: true,
    includeReferrer: true,
    trackingOptions: {
      city: false,
      ip_address: false,
    },
  };
  amplitude.getInstance().init(amplitudeKey, undefined, config);
  initialized = true;
}

initAmplitude();

function amplitudeLogger(name: string, values?: object) {
  if (!initialized) {
    console.log('Aktivitet uten initialisert amplitude:', name, values);
    return;
  }
  amplitude.getInstance().logEvent(name, values);
}

type AmplitudeAktivitetsData = {
  arbeidssoker: string;
  meldegruppe: string;
  innsendingstype: string;
};

export function loggAktivitet(
  aktivitet: string,
  data?: AmplitudeAktivitetsData
) {
  const eventData = { ...data, aktivitet: aktivitet };
  amplitudeLogger('meldekort.aktivitet', eventData);
}
