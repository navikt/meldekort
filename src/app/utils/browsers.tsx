// Edge
export const isEdge = window.navigator.userAgent.indexOf('Edge') !== -1;

// Internet Explorer 6-11
export const isIE =
  window.navigator.userAgent.indexOf('Trident') !== -1 && !isEdge;

// Firefox
export const isFireFox = window.navigator.userAgent.indexOf('Mozilla') !== -1;

// Safari
export const isSafari = window.navigator.userAgent.indexOf('Safari') !== -1;
