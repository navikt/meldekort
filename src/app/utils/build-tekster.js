const fs = require('fs');
const path = require('path');

function readFile(dir, file) {
    try {
        return fs.readFileSync(path.join(dir, file), 'utf8');
    } catch(e) {
        console.log('Error:', e.stack);
    }
}

function createJson(katalog) {
    const tekster = getNbAndEnTexts(katalog);
    return '/*tslint:disable*/export default {\'nb\':{' + tekster[0] + '},\'en\':{' + tekster[1] + '}};';
}

function getNbAndEnTexts(dir) {
    return read(dir, "", "");
}

function read(dir, nb, en) {
    let filename = "";
    fs.readdirSync(dir)
        .forEach((file) => {
            if (fs.statSync(path.join(dir, file)).isDirectory()) {
                [nb, en] = read(path.join(dir, file), nb, en);
            }
            else {
                if ( file.slice(-4) === '.txt') {
                    filename = "'" + file.split('_en.txt'|| '_nb.txt')[0] + "'" + ':' + "'" + readFile(dir, file).trim().replace(/\n|\r|\r\n/, ' ') + "',";
                    file.split('_')[1] === ('nb.txt' || 'AAP_nb.txt') ?
                        nb = nb + filename : en = en + filename;
                } else if (file.slice(-5) === '.html' && file.split('_nb.html')[0] === 'forklaring.sporsmal.aktivitet_arbeid_aap') {
                    console.log(file, ' ====== vår testfiler');

                }
                // Hvis ikke .txt, så html-fil.
                else {
                    console.log(file, ' er en html-fil.');
                }
            }
        });

    return [nb, en]
}

try{
    fs.writeFileSync('./src/app/tekster/alle-tekster.ts', createJson('./src/app/tekster/hjelpetekster'));
}catch (e){
    console.log("Kunne ikke skrive fil ", e);
}

console.log( getNbAndEnTexts('./src/app/tekster/hjelpetekster'));