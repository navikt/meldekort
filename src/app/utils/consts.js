const Konstanter = () => {
    return {
        hentHistoriskeMeldekortApiUri: 'person/historiskemeldekort',
        hentMeldekortdetaljerApiUri: 'meldekort/{id}',
        hentPersonStatusApiUri: 'person/status',
        hentMeldekortApiUri: 'person/meldekort',
        sendMeldekortApiUri: 'person/meldekort',
        sendMeldeformApiUri: 'person/meldeform',
        hentKorrigertMeldekortIdApiUri: 'meldekort/{id}/korrigering',
        metodeGet: 'GET',
        metodePost: 'POST'
    };
};

export default Konstanter;