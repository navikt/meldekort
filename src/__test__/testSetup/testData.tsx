import { MenyPunkt } from '../../app/utils/menyConfig';
import SendMeldekort from '../../app/sider/sendMeldekort/sendMeldekort';
import TidligereMeldekort from '../../app/sider/tidligereMeldekort/tidligereMeldekort';
import {
  KortStatus,
  KortType,
  Meldegruppe,
  Meldekort,
  MeldekortDag,
  Meldeperiode,
} from '../../app/types/meldekort';
import {
  Feilmelding,
  InnsendingState,
  UtfyllingFeil,
} from '../../app/types/innsending';
import { hentSporsmalConfig } from '../../app/sider/innsending/1-sporsmalsside/sporsmal/sporsmalConfig';
import { hentUtfyltDagConfig } from '../../app/sider/innsending/2-utfyllingsside/utfylling/utfyltDagConfig';

export const menyPunkterTEST: MenyPunkt[] = [
  {
    exact: true,
    component: SendMeldekort,
    tittel: 'sendMeldekort',
    tekstid: 'naviger.send',
    urlparam: '/send-meldekort',
    disabled: false,
  },
  {
    component: TidligereMeldekort,
    tekstid: 'sekundarmeny.tidligere',
    tittel: 'tidligereMeldekort',
    urlparam: '/tidligere-meldekort',
    disabled: false,
  },
];

export const mkDagTEST: MeldekortDag = {
  dag: 1,
  arbeidetTimerSum: 7.5,
  syk: false,
  annetFravaer: true,
  kurs: false,
};

const dateString = '2019-01-01T10:00:00Z';

export const mkDetaljerTEST = {
  id: '',
  meldekortId: 0,
  meldeperiode: '',
  arkivnokkel: '',
  kortType: KortType.KORRIGERT_ELEKTRONISK,
  meldeDato: new Date(dateString),
  lestDato: new Date(dateString),
  sporsmal: {
    annetFravaer: false,
    arbeidet: false,
    arbeidssoker: false,
    syk: false,
    kurs: false,
    signatur: false,
    meldekortDager: [],
  },
  begrunnelse: '',
};

export const sporsmalOgSvarTest = [
  {
    kategori: 'arbeid',
    sporsmal: 'sporsmal.arbeid',
    forklaring: 'forklaring.sporsmal.arbeid',
    svar: true,
    formatertDato: '20.01.2019',
  },
  {
    kategori: 'aktivitetArbeid',
    sporsmal: 'sporsmal.aktivitetArbeid',
    forklaring: 'forklaring.sporsmal.aktivitetArbeid',
    svar: false,
    formatertDato: '20.01.2019',
  },
];

const meldeperiodeTEST: Meldeperiode = {
  til: new Date(dateString),
  fra: new Date(dateString),
  kanKortSendes: true,
  kortKanSendesFra: new Date(dateString),
  periodeKode: '201901',
};

export const mkTEST: Meldekort = {
  meldekortId: 1,
  kortType: KortType.ELEKTRONISK,
  meldeperiode: meldeperiodeTEST,
  meldegruppe: Meldegruppe.DAGP,
  kortStatus: KortStatus.OPPRE,
  bruttoBelop: 0,
  mottattDato: new Date(dateString),
  korrigerbart: true,
};

export const mkDagerTEST: MeldekortDag[] = [
  {
    dag: 1,
    arbeidetTimerSum: 7.5,
    syk: false,
    annetFravaer: true,
    kurs: false,
  },
];

const feilmelding: Feilmelding = {
  feil: true,
  feilmelding: 'Test',
};

export const utfyllingFeilTEST: UtfyllingFeil = {
  feilIArbeid: feilmelding,
  feilIKurs: feilmelding,
  feilISyk: feilmelding,
  feilIFerie: feilmelding,
  feilKombinasjonDagpSykArbeid: false,
  feilKombinasjonDagpFravaerArbeid: false,
  feilKombinasjonAAPFravaerSyk: false,
  feilKombinasjonAAPFravaerArbeid: false,
  feilIArbeidetTimerHeleHalve: true,
  feilIArbeidetTimer: true,
  feilIDager: ['mandag'],
};

export const innsendingTEST: InnsendingState = {
  meldekortId: 0,
  korrigertMeldekortId: 0,
  innsendingstype: null,
  begrunnelse: {
    valgtArsak: '',
    erFeil: false,
  },
  sporsmalsobjekter: hentSporsmalConfig(),
  utfylteDager: hentUtfyltDagConfig(),
  meldekortdetaljer: {
    id: '',
    meldekortId: 0,
    meldeperiode: '',
    arkivnokkel: '',
    kortType: KortType.KORRIGERT_ELEKTRONISK,
    meldeDato: new Date(dateString),
    lestDato: new Date(dateString),
    sporsmal: {
      annetFravaer: false,
      arbeidet: false,
      arbeidssoker: false,
      syk: false,
      kurs: false,
      signatur: false,
      meldekortDager: [],
    },
    begrunnelse: '',
  },
  meldekortdetaljerInnsending: undefined,
  valideringsResultat: undefined,
};
