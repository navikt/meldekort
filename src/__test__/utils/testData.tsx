import { MenyPunkt } from '../../app/utils/menyConfig';
import SendMeldekort from '../../app/sider/sendMeldekort/sendMeldekort';
import TidligereMeldekort from '../../app/sider/tidligereMeldekort/tidligereMeldekort';

export const menyPunkterTEST: MenyPunkt[] = [
  {
    exact: true,
    component: SendMeldekort,
    tittel: 'sendMeldekort',
    tekstid: 'naviger.send',
    urlparam: '/send-meldekort',
    disabled: false,
  },
  {
    component: TidligereMeldekort,
    tekstid: 'sekundarmeny.tidligere',
    tittel: 'tidligereMeldekort',
    urlparam: '/tidligere-meldekort',
    disabled: false,
  },
];
