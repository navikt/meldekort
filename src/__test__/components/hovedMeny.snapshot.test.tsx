import * as React from 'react';
import * as renderer from 'react-test-renderer';
import ProviderWrapper from '../utils/providerWrapper';
import HovedMeny from '../../app/components/meny/desktop/hovedMeny';
import { menyPunkterTEST } from '../utils/testData';

it('HovedMeny', () => {
  const tree = renderer.create(
    <ProviderWrapper>
      <HovedMeny menypunkter={menyPunkterTEST} />
    </ProviderWrapper>
  );
  expect(tree).toMatchSnapshot();
});
