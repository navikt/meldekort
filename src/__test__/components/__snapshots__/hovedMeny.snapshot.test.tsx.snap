// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HovedMeny 1`] = `
<nav
  className="hovedmeny"
>
  <ul
    className="hovedmeny__wrapper"
  >
    <li
      className="hovedmeny__item"
    >
      <a
        aria-labelledby="tab hovedmeny__menypunkt"
        className="lenke hovedmeny__menypunkt active"
        href="#"
        onClick={[Function]}
      >
        Send 
      </a>
    </li>
    <li
      className="hovedmeny__item"
    >
      <a
        aria-labelledby="tab hovedmeny__menypunkt"
        className="lenke hovedmeny__menypunkt"
        href="#"
        onClick={[Function]}
      >
        Tidligere meldekort 
      </a>
    </li>
  </ul>
</nav>
`;
